dashboard/app/controllers/registrations_controller.rb

```diff
@@ -91,6 +96,109 @@ def upgrade
     respond_to_account_update(successfully_updated, success_message_kind)
   end
 
+  #
+  # PATCH /users/email
+  #
+  # Route allowing user to update their primary email address.
+  #
+  def set_email
+    return head(:bad_request) if params[:user].nil?
+    current_user.reload # Needed to make tests pass for reasons noted in set_email_test.rb
+
+    permitted = params.
+        require(:user).
+        permit(
+          :email,
+          :hashed_email,
+          :email_opt_in,
+          :current_password
+        )
```

Comment:
style: not a big deal, but typically these strong parameter declarations are in (private or protected) methods, e.g. 

```ruby
def email_params
  params.require(:user).
    permit(
      :email,
      :hashed_email,
      :email_opt_in,
      :current_password
    )
end
```

See http://api.rubyonrails.org/classes/ActionController/StrongParameters.html

Reply:
TIL!  Thanks!